### PR TITLE
Do not fail in JWT decode() if at_hash claim is missing

### LINKED ISSUE #75: Should at_hash claim verification fail when missing from JWT?
It looks like `at_hash` in JWT payload is optional (see http://openid.net/specs/openid-connect-core-1_0.html#CodeIDToken).

However, in python-jose, when both `id_token` and `access_token` parameters are specified, decoding a JWT that has no `at_hash` claim raises an error (*at_hash claim missing from token*)
https://github.com/mpdavis/python-jose/pull/30/files#diff-b106d01229785c64375df96ca4b3f58cR422

Shouldn't it be acceptable since the spec says it's optional?

Obviously we can disable at_hash verification with the appropriate decode option, but we find it useful to perform claims verification on JWT that have it or not with the same code. Maybe with a `allow_missing_at_hash` option or something?

Huge thanks for this lib 😻

### PR DESCRIPTION
**Pull request by leplatrem**
Fixes #75

### CODE REVIEW

File: jose/jwt.py [hunk 1]
```diff
@@ -418,12 +418,11 @@ def _validate_at_hash(claims, access_token, algorithm):
     """
     if 'at_hash' not in claims and not access_token:
         return
+    elif access_token and 'at_hash' not in claims:
```
Comment (mpdavis) [inline jose/jwt.py]: We can refactor this logic for simplification:

```python
if 'at_hash' not in claims:
    return

if not access_token:
    msg = 'No access_token provided to compare against at_hash claim.'
    raise JWTClaimsError(msg)
```

File: jose/jwt.py [hunk 2]
```diff
@@ -418,12 +418,11 @@ def _validate_at_hash(claims, access_token, algorithm):
     """
```
Comment (mpdavis) [inline jose/jwt.py]: Nit: Can you update the docstring for this method to pull the explanation from the RFC, similar to the other validation methods?

```
    """
    Validates that the 'at_hash' is valid.
    
    Its value is the base64url encoding of the left-most half of the hash
    of the octets of the ASCII representation of the access_token value,
    where the hash algorithm used is the hash algorithm used in the alg
    Header Parameter of the ID Token's JOSE Header. For instance, if the
    alg is RS256, hash the access_token value with SHA-256, then take the
    left-most 128 bits and base64url encode them. The at_hash value is a
    case sensitive string.  Use of this claim is OPTIONAL.

    Args:
        claims (dict): The claims dictionary to validate.
        access_token (str): The access token returned by the OpenID Provider.
        algorithm (str): The algorithm used to sign the JWT, as specified by
            the token headers.
    """
```

**Comment by mpdavis** [comment]
It looks like there are builds failing on Python 2.6 and 3.3.

Python 2.6 is failing because pytest dropped support. It looks like we will need to pin pytest in `tox.ini` for 2.6 builds (or possibly just all builds if easier).

I am still looking into the 3.3 failure. I can take a look later if you don't want to worry about it.

**Comment by leplatrem** [comment]
I also saw this:
```
$ tox

Matching undeclared envs is deprecated. Be sure all the envs that Tox should run are declared in the tox config.
```

**Comment by leplatrem** [comment]
@mpdavis  This should be ready to land :)

**Comment by zejn** [comment]
The Python 2.6 has been unsupported for a long time now and 3.3 is also unsupported since late 2017. 

App engine has Python 2.7, so this looks good to me.

**Comment by Integralist** [comment]
What came of this? I've recently discovered this issue myself and would be keen to see a fix land. Thanks all for their work on this 👍

**Comment by techdragon** [comment]
Is this still problematic? Or can this just be merged, I've noticed this is the root cause of some issues in my deployed applications, I'd rather not work around such a simple, and already written, fix.

**Comment by rcjsuen** [comment]
I hope this can eventually land at some point. In the meantime, thank you @leplatrem for finding the issue and suggesting a fix!

**Comment by mpdavis** [comment]
lgtm

**Pull request merged**